Eastern-Research-Group/mywaterway
# Feature/186 locations loss

## Related Issues:
* [HMW-186](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=864&view=detail&selectedIssue=HMW-186&quickFilter=2479)

## Main Changes:
* Moved a `useEffect` hook that plotted streamgages into a `useCallback` function, which is then called from a closely related effect. Altering the streamgage data after plotting or too many calls to `plotGages` was causing features on the map to drop off when switching tabs.
* Added a flag to the `MapHighlightContext` to signal when the feature cache employed by `useWaterbodyHighlight` goes stale. Re-using the cache between the **Overview** and **Monitoring** tabs resulted in features not being highlighted.
* Added an optional prop to the `VirtualizedList` component that lists which type of data is displayed in the list. This was done to fix a problem in the **Monitoring & Sensors** list in the **Overview** tab: if only *Current Water Conditions* was initially toggled on, then *Sample Locations*, the list would not resize to the larger row height of the monitoring locations accordion items, so the prop is used to signal a resize.

## Steps To Test:
1. Navigate to the **Overview** tab of the **Community** page
2. Click the *Monitoring & Sensors* sub-tab
3. Toggle on the *Current Water Condition*, and highlight a few of the features
4. Go to the **Monitoring** tab
5. Confirm that all features are still visible, and that highlighting works properly
